### PR TITLE
Update catch2

### PR DESCRIPTION
**Pull request by PDoakORNL**
## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## What type(s) of changes does this code introduce?

This updates catch2 to the newest release (2.13.6).  Most importantly this brings a micro benchmarking framework to catch2.  It includes updated complex_approx .hpp which as far as I can tell is a qmcpack addition to catch2 and only needed to be moved the the detail namespace.

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Leconte

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes/No. This PR is up to date with current the current state of 'develop'
- Yes/No. Code added or changed in the PR has been clang-formatted
- Yes/No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes/No. Documentation has been added (if appropriate)

### CODE REVIEW

File: external_codes/catch/complex_approx.hpp [hunk 1]
```diff
@@ -87,20 +88,22 @@ class ComplexApprox
...
 template<>
-struct StringMaker<ComplexApprox> {
-  static std::string convert(ComplexApprox const &value);
+struct StringMaker<Catch::Detail::ComplexApprox> {
+  static std::string convert(Catch::Detail::ComplexApprox const &value);
 };
 
 #ifdef CATCH_IMPL
-std::string StringMaker<ComplexApprox>::convert(ComplexApprox const& value)
+  std::string StringMaker<Catch::Detail::ComplexApprox>::convert(Catch::Detail::ComplexApprox const& value)
```
Comment (ye-luo) [inline external_codes/catch/complex_approx.hpp:99]: Formatting?

**Comment by markdewing** [comment]
`catch.hpp` should include `complex_approx.hpp` and `log_complex_approx.hpp` at the end.  Unless there's another solution to get these files included.

**Review by prckent — COMMENTED** [review]
Good to update and benefit. The complex_approx was the main (only?) change I recall. git blame may reveal more. Edit: Mark replied above adding log_complex_approx

Drive-by comment that shouldn't hold up this PR: What is the purpose of src/Configuration.h exactly? Why does it define QMCPLUSPLUS_TRAITS_H and not CONFIGURATION?... Is top level src/ really the appropriate location?

**Comment by PDoakORNL** [comment]
I found that including them at the end didn't work anymore but they need to be included right after the Approx definition, see line 3191

File: external_codes/catch/complex_approx.hpp (hunk 1 above)
Comment (PDoakORNL) [reply external_codes/catch/complex_approx.hpp]: done

**Comment by PDoakORNL** [comment]
now that I try it again that does work, but it makes more sense to me for them to be included immediately after approx. There are many ifdefs closing at the end of the file that don't have anything to do with Approx.  Also shouldn't we try to get these included into catch2?

**Review by markdewing — APPROVED** [review]

**Pull request merged**